src/write_gcm.c

```diff
@@ -1412,7 +1415,8 @@ static void wg_payload_destroy(wg_payload_t *list) {
 // us remember that it is a compile-time string constant which does not need to
 // be copied/deallocated.
 static int wg_typed_value_create_from_value_t_inline(wg_typed_value_t *result,
-    int ds_type, value_t value, const char **dataSourceType_static) {
+    int ds_type, char *metric_value_type, value_t value, 
```

Comment:
I'd s/`metric_value_type`/`value_type`/. Also, why not make it an `enum` instead of a string (i.e., parse the string earlier)? That way you can also add a static array mapping these `enum` values (e.g., `INT64` and `DOUBLE`) into triples of `"int64value"`/ `"doubleValue"`, `"INT64"`/`"DOUBLE"`, and the corresponding format strings...